support-frontend/assets/pages/contributions-landing/components/PaymentMethodSelector.tsx

```diff
@@ -34,61 +39,48 @@ import {
 	Sepa,
 	Stripe,
 } from 'helpers/forms/paymentMethods';
+import type { Switches } from 'helpers/globalsAndSwitches/settings';
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import { getReauthenticateUrl } from 'helpers/urls/externalLinks';
+import { classNameWithModifiers } from 'helpers/utilities/utilities';
 import type { Action } from '../contributionsLandingActions';
 import {
 	loadAmazonPaySdk,
 	loadPayPalExpressSdk,
 	updatePaymentMethod,
 	updateSelectedExistingPaymentMethod,
 } from '../contributionsLandingActions';
-import {
-	getExistingPaymentMethodLabel,
-	isUsableExistingPaymentMethod,
-	mapExistingPaymentMethodToPaymentMethod,
-	subscriptionsToExplainerList,
-	subscriptionToExplainerPart,
-} from 'helpers/forms/existingPaymentMethods/existingPaymentMethods';
-import type {
-	ExistingPaymentMethod,
-	RecentlySignedInExistingPaymentMethod,
-} from 'helpers/forms/existingPaymentMethods/existingPaymentMethods';
-import { getReauthenticateUrl } from 'helpers/urls/externalLinks';
-import { Radio, RadioGroup } from '@guardian/src-radio';
-import SvgSepa from 'components/svgs/sepa';
-import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import type { State } from '../contributionsLandingReducer';
 import ContributionChoicesHeader from './ContributionChoicesHeader';
-// ----- Types ----- //
 
-/* eslint-disable react/no-unused-prop-types */
-type PropTypes = {
+// ----- Component ----- //
+
+interface PaymentMethodSelectorProps {
 	countryId: IsoCountry;
 	countryGroupId: CountryGroupId;
 	contributionType: ContributionType;
 	currency: IsoCurrency;
-	existingPaymentMethods: ExistingPaymentMethod[] | typeof undefined;
+	existingPaymentMethods?: ExistingPaymentMethod[];
 	paymentMethod: PaymentMethod;
-	existingPaymentMethod: RecentlySignedInExistingPaymentMethod;
-	updatePaymentMethod: (
-		arg0: PaymentMethod,
-	) => (arg0: (...args: any[]) => any) => void;
+	existingPaymentMethod?: RecentlySignedInExistingPaymentMethod;
+	updatePaymentMethod: (paymentMethod: PaymentMethod) => void;
 	updateSelectedExistingPaymentMethod: (
-		arg0: RecentlySignedInExistingPaymentMethod | typeof undefined,
+		existingPaymentMethod?: RecentlySignedInExistingPaymentMethod,
 	) => Action;
 	isTestUser: boolean;
 	switches: Switches;
 	payPalHasBegunLoading: boolean;
 	amazonPayHasBegunLoading: boolean;
-	loadPayPalExpressSdk: (
-		contributionType: ContributionType,
-	) => (dispatch: (...args: any[]) => any) => void;
+	loadPayPalExpressSdk: (contributionType: ContributionType) => void;
 	loadAmazonPaySdk: (
 		countryGroupId: CountryGroupId,
 		isTestUser: boolean,
-	) => (dispatch: (...args: any[]) => any) => void;
+	) => void;
```

Comment:
I've seen this a few times in the `flow` code. We were typing explicitly the fact that these action creators are actually thunk action creators. But all that detail is actually hidden from the component, the function signature it gets is much simpler!